### ISSUE TITLE
blogTitle not working

### ISSUE THREAD

**Issue by root-io** [facebook/docusaurus#10045]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [ ] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

blogTitle not working

Someone at the end of this thread seems to have a similar issue: https://github.com/facebook/docusaurus/discussions/5119

### Steps to reproduce

```typescript
  presets: [
    [
      'classic',
      {
        blog: {
          routeBasePath: "/",
          blogTitle: "Blog",
        },
      } satisfies Preset.Options,
    ],
  ],
```

### Expected behavior

I want the title of the page to be: "Blog | Foo"

### Actual behavior

The blogTitle is not showing, I get the config title twice: "Foo | Foo".

### Your environment

- Public source code:
- Public site URL:
- Docusaurus version used: 3.2.1
- Environment name and version (e.g. Chrome 89, Node.js 16.4):
- Operating system and version (e.g. Ubuntu 20.04.2 LTS):

### Self-service

- [ ] I'd be willing to fix this bug myself.

**Comment by OzakIOne** [comment]
Hello I tried when creating a minimal repo and I can't reproduce your issue

https://stackblitz.com/edit/github-srzyqm?file=docusaurus.config.js

![image](https://github.com/facebook/docusaurus/assets/29860391/a7fcebd5-24ff-4364-8018-c6973aa9f727)

**Comment by root-io** [comment]
Hey OzaklOne thanks for the stackblitz, I tried running it but I get the following error: `error An unexpected error occurred: "Failed to fetch dependencies: - @types/gtag.js@0.0.12".`

**Comment by OzakIOne** [comment]
From what I tried and understand, when you are in blog only mode `blogTitle` is not the title of the page anymore, blogTitle is the title of the page when `routeBasePath` is not set to `'/'` because `title` will take the priority over `blogTitle` and indeed on the route `/` of your website you will have title twice so by default `'My blog | My blog'`

**Comment by slorber** [comment]
Yes that looks like we have this historical behavior in the theme:

```tsx
function BlogListPageMetadata(props: Props): JSX.Element {
  const {metadata} = props;
  const {
    siteConfig: {title: siteTitle},
  } = useDocusaurusContext();
  const {blogDescription, blogTitle, permalink} = metadata;
  const isBlogOnlyMode = permalink === '/';
  const title = isBlogOnlyMode ? siteTitle : blogTitle;
  return (
    <>
      <PageMetadata title={title} description={blogDescription} />
      <SearchMetadata tag="blog_posts_list" />
    </>
  );
}
```

I'd consider it a bug because that looks wrong to me.

Having a blog homepage at `/` does not mean a site only contains a blog. You might have a blog at the root of your site and yet have docs.

I'm not sure how we'll fix it yet, but this will likely be done as part of a global siteTitle breaking-change PR where we change other things as well.

**Comment by kalyan90** [comment]
@slorber I would like to work on this, can you please help with the expected behaviour?

**Comment by japj** [comment]
I also just ran into this for my site, where a blog is hosted at '/', but it also contains some documentation under '/docs'.

I was expecting "Blog|my site", but got "my site|my site" instead as title.

**Comment by japj** [comment]
I am fairly new to docusaurus and react, but managed to swizzle eject BlogListPage (since BlogListMetadata seems to be part of that): https://github.com/japj/japj.github.io/commit/82835784e9f0a793c757bcbfb2cfa2833619c5a1#diff-a513ae010e0223ea74a6354e97b3df6476835c573e0bfa189283e2f5b3e5c9f5R23

Is this the 'correct' way, or is there a better way to do this?
